packages/nextjs/src/api/index.ts

```diff
@@ -1,6 +1,65 @@
-if (!process.env.CLERK_API_KEY) {
-  throw Error('The CLERK_API_KEY environment variable must be set to use imports from @clerk/nextjs/api.');
-}
+import {
+  ClerkMiddleware,
+  ClerkMiddlewareOptions,
+  default as Clerk,
+  RequireAuthProp,
+  WithAuthProp,
+} from '@clerk/clerk-sdk-node';
+import type { NextApiRequest, NextApiResponse } from 'next';
 
-export { default } from '@clerk/clerk-sdk-node';
+export default Clerk;
 export * from '@clerk/clerk-sdk-node';
+
+// https://nextjs.org/docs/api-routes/api-middlewares#connectexpress-middleware-support
+function runMiddleware(req: NextApiRequest, res: NextApiResponse, fn: ClerkMiddleware) {
+  return new Promise((resolve, reject) => {
+    // @ts-ignore
+    fn(req, res, result => {
+      if (result instanceof Error) {
+        return reject(result);
+      }
+      return resolve(result);
+    });
+  });
+}
+
+type NextApiHandlerWithAuth<T = any> = (
+  req: WithAuthProp<NextApiRequest>,
+  res: NextApiResponse<T>,
+) => void | Promise<void>;
+
+// Set the session on the request and then call provided handler
+export function withAuth(handler: NextApiHandlerWithAuth, options?: ClerkMiddlewareOptions): NextApiHandlerWithAuth {
+  return async (req, res) => {
+    await runMiddleware(req, res, Clerk.expressWithAuth(options));
+    return handler(req, res);
+  };
+}
+
+type NextApiHandlerRequireAuth<T = any> = (
+  req: RequireAuthProp<NextApiRequest>,
+  res: NextApiResponse<T>,
+) => void | Promise<void>;
+
+export function requireAuth(
+  handler: NextApiHandlerRequireAuth,
+  options?: ClerkMiddlewareOptions,
+): NextApiHandlerRequireAuth {
+  return async (req, res) => {
+    try {
+      await runMiddleware(req, res, Clerk.expressRequireAuth(options));
+      return handler(req, res);
+    } catch (error) {
+      // @ts-ignore
+      const errorData = error.data || { error: error.message };
+      // @ts-ignore
+      res.statusCode = error.statusCode || 401;
+      /**
+       * Res.json is available in express-like environments.
+       * Res.send is available in express-like but also Fastify.
```

Comment:
❓  
This will be a breaking change for Fastify clients right ? _The way we used it up until now can be seen [here](https://github.com/clerkinc/clerk-fastify-react-prisma-starter/blob/main/packages/server/src/auth/clerkHandler.ts)._

Reply:
Correct. Fastify clients will have to update to a new method coming from clerk-sdk-node